R/verbatim_popup.R

```diff
@@ -36,31 +41,25 @@ verbatim_popup_ui <- function(id, button_label, ...) {
 #' @param verbatim_content (`character`, `expression` or `reactive(1)` holding either)
 #' the content to show in the popup modal window
 #' @param title (`character(1)`) the title of the modal window
+#' @param style (`logical(1)`) whether to style the `verbatim_content` using `styler::style_text`
 #' @param disabled (`reactive(1)`) the `shiny` reactive value holding a `logical`. The popup button is disabled
 #' when the flag is `TRUE` and enabled otherwise
 #'
-verbatim_popup_srv <- function(id, verbatim_content, title, disabled = shiny::reactiveVal(FALSE)) {
+verbatim_popup_srv <- function(id, verbatim_content, title, style = FALSE, disabled = shiny::reactiveVal(FALSE)) {
   checkmate::assert_string(id)
   checkmate::assert_string(title)
+  checkmate::assert_flag(style)
   checkmate::assert_class(disabled, classes = "reactive")
-  if (inherits(verbatim_content, "reactive")) {
-    shiny::validate(shiny::need(
-      checkmate::assert_multi_class(verbatim_content(), classes = c("expression", "character"))
```

Comment:
`verbatim_content()` wasn't inside of the `reactive` - it was causing the failures